src/node/Main.hs

```diff
@@ -109,77 +121,139 @@ action
+                    wDhtWorkers = first (map $ wrapActionSpec $ "worker" <> "dht")
+                                        (dhtWorkers (rmGetPeers liftedTransport) kademliaInst)
+                let allPlugins :: ([WorkerSpec (StatsMode SscGodTossing)], OutSpecs)
+                    allPlugins = mconcat [ wDhtWorkers
+                                         , convPlugins currentPluginsGT
+                                         , utwStats]
+                runNodeStats @SscGodTossing
+                    (CLI.peerId commonArgs)
+                    liftedTransport
+                    allPlugins
+                    currentParams gtParams
+            (True, NistBeaconAlgo) -> do
+                let liftedTransport = hoistResources lift transport
+                    wDhtWorkers = first (map $ wrapActionSpec $ "worker" <> "dht")
+                                        (dhtWorkers (rmGetPeers liftedTransport) kademliaInst)
+                let allPlugins :: ([WorkerSpec (StatsMode SscNistBeacon)], OutSpecs)
+                    allPlugins = mconcat [ wDhtWorkers
+                                         , convPlugins currentPlugins
+                                         , utwStats]
+                runNodeStats @SscNistBeacon
+                    (CLI.peerId commonArgs)
+                    (hoistResources lift transport)
+                    allPlugins
+                    currentParams ()
+            (False, GodTossingAlgo) -> do
+                let liftedTransport = hoistResources lift transport
+                    wDhtWorkers = first (map $ wrapActionSpec $ "worker" <> "dht")
+                                        (dhtWorkers (rmGetPeers liftedTransport) kademliaInst)
+                let allPlugins :: ([WorkerSpec (ProductionMode SscGodTossing)], OutSpecs)
+                    allPlugins = mconcat [ wDhtWorkers
+                                         , convPlugins currentPluginsGT
+                                         , utwProd]
+                runNodeProduction @SscGodTossing
+                    (CLI.peerId commonArgs)
+                    liftedTransport
+                    allPlugins
+                    currentParams gtParams
+            (False, NistBeaconAlgo) -> do
+                let liftedTransport = hoistResources lift transport
+                    wDhtWorkers = first (map $ wrapActionSpec $ "worker" <> "dht")
+                                        (dhtWorkers (rmGetPeers liftedTransport) kademliaInst)
+                let allPlugins :: ([WorkerSpec (ProductionMode SscNistBeacon)], OutSpecs)
+                    allPlugins = mconcat [ wDhtWorkers
+                                         , convPlugins currentPlugins
+                                         , utwProd]
+                runNodeProduction @SscNistBeacon
+                    (CLI.peerId commonArgs)
+                    liftedTransport
+                    allPlugins
+                    currentParams ()
```

Comment:
It's unbelievably ugly, I don't think it will be possible to live if it gets merged to master.